server/channels/app/oauth.go

```diff
@@ -958,7 +958,14 @@ func (a *App) SwitchEmailToOAuth(c request.CTX, w http.ResponseWriter, r *http.R
 	stateProps["email"] = email
 
 	if service == model.UserAuthServiceSaml {
-		return a.GetSiteURL() + "/login/sso/saml?action=" + model.OAuthActionEmailToSSO + "&email=" + utils.URLEncode(email), nil
+		samlToken, samlErr := a.CreateSamlRelayToken(email)
+		if samlErr != nil {
+			return "", samlErr
+		}
+
+		token := b64.StdEncoding.EncodeToString([]byte(samlToken.Token))
```

Comment:
Why the base64? The token is already zbase32, isn't it?

Reply:
True, fixed